### PR TITLE
Support ARM SVE via vector_size API

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

File: simd.hpp [hunk 1]
```diff
@@ -49,9 +49,7 @@
 
 #include "pack.hpp"
 
-#if defined(__clang_major__) && (__clang_major__ >= 11)
```
Comment (crtrott) [inline simd.hpp:52]: Moved the correct guard into the vector_size.hpp file

File: simd.hpp [hunk 2]
```diff
@@ -97,12 +95,16 @@ using native = avx512;
 using native = avx;
 #elif defined(__SSE2__)
 using native = sse;
-#elif defined(__ARM_NEON)
+#elif defined(__ARM_NEON) && !defined(__ARM_FEATURE_SVE_BITS)
```
Comment (crtrott) [inline simd.hpp:98]: NEON is defined for SVE supported archs too, but we don't want to use it as the native type.

File: simd.hpp [hunk 3]
```diff
@@ -97,12 +95,16 @@ using native = avx512;
...
 using native = neon;
 #elif defined(__VSX__)
 using native = vsx;
 #else
-using native = pack<8>;
+#if defined(__ARM_FEATURE_SVE_BITS)
+using native = vector_size<__ARM_FEATURE_SVE_BITS/8>;
```
Comment (crtrott) [inline simd.hpp:104]: vector size takes the size in bytes.

File: vector_size.hpp [hunk 4]
```diff
@@ -45,7 +45,8 @@
 
 #include "simd_common.hpp"
 
-#if defined(__clang__)
+#if (defined(__clang__) && (__clang_major__ >= 11)) || \
+    (defined(__GNUC__) && (__GNUC__ >= 10) && (__GNUC_MINOR__ >= 2))
```
Comment (crtrott) [inline vector_size.hpp:49]: GNU 10.2 supports this

**Review by DavidPoliakoff — APPROVED** [review]
Looks good to me, certainly supports the SVE that we know about

**Pull request merged**